Modules/_io/bufferedio.c

```diff
@@ -1767,6 +1778,27 @@ _bufferedwriter_reset_buf(buffered *self)
     self->write_end = -1;
 }
 
+static void
+_bufferedwriter_set_append(buffered *self)
+{
+    PyObject *mode;
+
+    mode = _PyObject_GetAttrId(self->raw, &PyId_mode);
+    if (mode != NULL) {
+        /* Raw fileobj has no mode string so as far as we can know it has
+           normal write behavior */
+        if (PyUnicode_FindChar(mode, 'a', 0, PyUnicode_GET_LENGTH(mode), 1) != -1) {
+            self->appending = 1;
+        } else {
+            self->appending = 0;
+        }
+        Py_DECREF(mode);
+    } else {
+        PyErr_Clear();
```

Comment:
Exceptions like `MemoryError` or `KeyboardInterrupt` should not be cleared. `PyErr_ExceptionMatches()` could be used instead of blindly clearing any exception.

Reply:
The original patch for this was written so long ago, I don't even remember why there's a `PyErr_Clear()` there, but I don't think it's even necessary.

Reply:
Now I remember why that was there.  Fixed in 08661e3